internal/pkg/cli/pipeline_deploy.go

```diff
@@ -148,19 +150,34 @@ func newDeployPipelineOpts(vars deployPipelineVars) (*deployPipelineOpts, error)
 	}, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *deployPipelineOpts) Validate() error {
+	return nil
+}
+
+// Ask prompts the user for any unprovided required fields and validates them.
+func (o *deployPipelineOpts) Ask() error {
+	if o.appName != "" {
+		if _, err := o.store.GetApplication(o.appName); err != nil {
+			return err
+		}
+	} else {
+		if err := o.askAppName(); err != nil {
+			return err
+		}
+	}
```

Comment:
We shouldn't ask for the app name as the command has to be run within the workspace. Otherwise we won't be able to find the correct workspace. Please refer to https://github.com/aws/copilot-cli/pull/3317/files#diff-1e7f78366fc8d6eec3339ca563e5fcbe1226766d42a4d3a02b730d9fbd2cc5a7R174

Reply:
Yup, you're right! Fixing.